libs/priors/bbhnet/priors/priors.py

```diff
@@ -1,101 +1,66 @@
-
-    Returns:
-        grid:
-            The values at which the pdf was estimated
-        pdf:
-            The estimated pdf
-    """
-    param_min = np.min(param_values)
-    param_max = np.max(param_values)
-    if spacing == "lin":
-        bins = np.linspace(param_min, param_max, grid_size + 1)
-        grid = (bins[:-1] + bins[1:]) / 2
-    elif spacing == "log":
-        min_exp = np.log10(param_min)
-        max_exp = np.log10(param_max)
-        bins = np.logspace(min_exp, max_exp, grid_size + 1)
-        grid = np.sqrt(bins[:-1] * bins[1:])
-    else:
-        raise ValueError("Spacing must be either 'lin' or 'log'")
-
-    pdf, _ = np.histogram(param_values, bins, density=True)
-
-    return grid, pdf
-
-
-def read_priors_from_file(event_file: Path, *parameters: str) -> BBHPriorDict:
-    """
-    Reads in a file containing sets of GW parameters and
-    returns a set of interpolated priors
-    The expected structure is based off the event file from
-    here: https://dcc.ligo.org/T2100512
-
-    Args:
-        event_file: An hdf5 file containing event parameters
-        parameters: Optional, a list of parameters to read from the file
-
-    Returns:
-        prior: A BBHPriorDict with priors based on the event file
-    """
-    prior = BBHPriorDict()
-    with h5py.File(event_file, "r") as f:
-        events = f["events"]
-        field_names = parameters or events.dtype.names
-        for name in field_names:
-            grid, pdf = pdf_from_events(events[name])
-            prior[name] = Interped(grid, pdf, np.min(grid), np.max(grid))
-
-    return prior
-
-
-def uniform_extrinsic() -> BBHPriorDict:
-    prior = BBHPriorDict()
+class PriorDict(BBHPriorDict):
+    def __init__(self, *args, source_frame: bool = False, **kwargs):
+        super().__init__(*args, **kwargs)
+        # whether the priors are defined in source frame
+        self._source_frame = source_frame
+
+    def sample(self, n: Optional[int] = None, source_frame: bool = False):
+        samples = super().sample(n)
+        # only rescale if requested frame is different from defined frame
+        rescale = self._source_frame or source_frame
+        if not rescale:
+            return samples
+
+        # convert to requested frame
+        factor = 1 + samples["redshift"]
+        factor = (1 / factor) if source_frame else factor
+        samples["mass_1"] *= factor
+        samples["mass_2"] *= factor
+        return samples
+
+    def prob(self, samples, source_frame: bool = False):
+        """
+        Override the default prob method to account for the jacobian
+        between detector frame and source frame masses.
+        See equation 7) in https://arxiv.org/pdf/1712.00482.pdf
+        """
+        prob = super().prob(samples)
+        rescale = self._source_frame or source_frame
+        if not rescale:
+            return prob
+
+        factor = (1 + samples["redshift"]) ** 2
+        jacobian = factor if source_frame else (1 / factor)
```

Comment:
Since we define our priors with redshift I don't think we need that factor. 
If we were to have defined our prior in luminosity distance and then converted to redshift, that factor would matter.

Reply:
Ah, gotcha, that makes sense